### PR TITLE
Fix 2 build errors.

### PR DESCRIPTION
**Pull request by baohaojun**

### CODE REVIEW

**Comment by axel-h** [comment]
The fix https://github.com/seL4/global-components/pull/21/commits/f086551cd5bde3f68936e15fee0512618879cf09 seem trivial to ger merged, this should be cherry-picked. I'm not sure about the other change.

File: templates/seL4RPCDataport-to.template.h [hunk 1]
```diff
@@ -3,7 +3,6 @@
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
-  #*/
```
Comment (lsf37) [inline templates/seL4RPCDataport-to.template.h:6]: This one is not a preprocessor directive, it's a template comment that has gone wrong (probably may fault). It should be

```
/*#
 * ..
#*/
```

Comment (kent-mcleod) [reply templates/seL4RPCDataport-to.template.h:6]: It was already fixed in https://github.com/seL4/global-components/commit/02133497ed126910ec0f4ddb98b997e481d91ffa I believe

Comment (lsf37) [reply templates/seL4RPCDataport-to.template.h:6]: Ah, yes, hence the conflicts.

**Comment by lsf37** [comment]
@kent-mcleod are you happy with the other commit? If yes, we should rebase and merge that one.

**Comment by kent-mcleod** [comment]
> @kent-mcleod are you happy with the other commit? If yes, we should rebase and merge that one.

I don't believe the other commit is correct sorry.  My understanding is that the actual build issue is related to `picotcp-socket-sync-server` depending on `picotcp` but due to the way `picotcp`is declared in `util_libs`. If the config options `LibPicotcp` or `LibPicotcpBsd` when the build file in util_libs/picotcp is processed then `picotcp` won't be declared and the include paths for picotcp will not be provided during compilation.

Currently making changes like https://github.com/seL4/camkes/pull/18 are required to ensure that `LibPicotcp`is set early enough in the build config. But a long-term solution would be to rewrite the picotcp CMakeLists.txt to not require `LibPicotcp` or `LibPicotcpBsd` at all but this is more work.

**Comment by lsf37** [comment]
Thanks for that. So seL4/camkes#18 would fix the build issue this PR is for?

**Comment by kent-mcleod** [comment]
> Thanks for that. So [seL4/camkes#18](https://github.com/seL4/camkes/pull/18) would fix the build issue this PR is for?

I don't think so because that app doesn't use the `picotcp-socket-sync` module that this PR is changing.  But I think it does address the last picotcp example that we have with this issue.

**Comment by baohaojun** [comment]
Thanks, the picoserver build error should have been fixed by https://github.com/seL4/camkes/pull/18/commits/89ff0f1f311944b75f2102b0fbbdf2b891878afe, hence close this pull request.

**Pull request closed without merging**